Parser/tokenizer.c

```diff
@@ -370,6 +370,57 @@ update_fstring_buffers(struct tok_state *tok, char value, int regular, int multi
     }
 }
 
+static int
+update_fstring_expr(struct tok_state *tok, char cur)
+{
+    if (!tok->cur || !tok->tok_mode_stack_index) {
+        return 1;
+    }
+
+    Py_ssize_t size = strlen(tok->cur);
```

Comment:
Hummmm, i think this is correct but I am not sure if the buffer is always 0 terminated. We should check that that is true.